src/ui/controls/color-picker/hsb-picker.tsx

```diff
@@ -118,15 +121,24 @@ export default function HSBPicker(props: HSBPickerProps) {
     }
 
     function onSBCanvasRender(canvas: HTMLCanvasElement) {
+        if (isElementHidden(canvas)) {
+            context.store.wasPrevHidden = true;
+            return;
+        }
         const hue = activeHSB.h;
         const prevHue = prevColor && rgbToHSB(parse(prevColor)).h;
-        if (hue === prevHue) {
+        if (!context.store.wasPrevHidden && hue === prevHue) {
```

Comment:
Note sure, but maybe it should be `||` instead of `&&`?

Reply:
Both ways works the same way, but I can see a very very unlikely situation where `||` is suitable.

Reply:
Actually - I was wrong, and having `||` wouldn't render the hue on the first time.

Reply:
Because technically `hue === prevHue` is correct, because it wasn't rendered yet.